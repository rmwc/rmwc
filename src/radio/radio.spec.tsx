import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Radio } from './';

describe('Radio', () => {
  test('renders', () => {
    const { container } = render(<Radio label="test" />);
    expect(screen.getByText('test')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('can be checked', () => {
    render(<Radio checked onChange={() => {}} />);
    expect(screen.getByRole('radio')).toHaveAttribute('checked');
  });

  test('handles onChange', () => {
    let val = 0;
    render(<Radio checked onChange={() => val++} label="Click me" />);

    userEvent.click(screen.getByText('Click me'));

    waitFor(() => {
      expect(val).toBe(1);
    });
  });

  test('can be disabled', () => {
    render(<Radio disabled />);

    expect(screen.getByRole('radio')).toHaveAttribute('disabled');
  });

  test('can have custom classnames on input', () => {
    const { container } = render(<Radio className={'my-custom-classname'} />);
    expect(container.firstChild).toHaveClass('my-custom-classname');
  });
});
