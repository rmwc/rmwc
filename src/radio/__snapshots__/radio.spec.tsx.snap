// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders 1`] = `
<div>
  <div
    class="mdc-form-field"
  >
    <div
      class="mdc-ripple-upgraded--unbounded mdc-radio"
      data-mdc-ripple-is-unbounded="true"
    >
      <input
        class="mdc-radio__native-control"
        id="toggle--test"
        type="radio"
      />
      <div
        class="mdc-radio__background"
      >
        <div
          class="mdc-radio__outer-circle"
        />
        <div
          class="mdc-radio__inner-circle"
        />
      </div>
      <div
        class="mdc-radio__ripple"
      />
    </div>
    <label
      for="toggle--test"
      id="toggle--testlabel"
    >
      test
    </label>
  </div>
</div>
`;
